api_docs/server_doc_gen.py

```diff
@@ -0,0 +1,118 @@
+"""API Doc generator for the server side API."""
+
+from doc_writer import HydraDoc, HydraClass, HydraClassProp, HydraClassOp
+import json
+
+
+def server_doc(API, BASE_URL):
+    """Generate API Doc for server."""
+    # Main API Doc
+    api_doc = HydraDoc(API,
+                       "API Doc for the server side API",
+                       "API Documentation for the server side system",
+                       API,
+                       BASE_URL)
+
+    # Drone operations on server must be linked to same operations on the specific drone
+    status = HydraClass("Status", "Status", "Class for drone status objects")
+    # Properties
+    status.add_supported_prop(HydraClassProp("http://auto.schema.org/speed", "Speed", True, False, False))
+    status.add_supported_prop(HydraClassProp("http://schema.org/geo", "Position", True, False, False))
+    status.add_supported_prop(HydraClassProp("http://schema.org/fuelCapacity", "Battery", True, True, False))
+    status.add_supported_prop(HydraClassProp("http://schema.org/device", "Sensor", True, True, False))
+    status.add_supported_prop(HydraClassProp("http://schema.org/model", "Model", True, True, False))
+    status.add_supported_prop(HydraClassProp("https://schema.org/status", "SensorStatus", True, False, False))
+
+    # Drone Class
+    drone = HydraClass("Drone", "Drone", "Class for a drone")
+    # Properties
+    drone.add_supported_prop(HydraClassProp("vocab:Status", "DroneStatus", True, False, False))
+    drone.add_supported_prop(HydraClassProp("http://schema.org/name", "name", True, False, False))
+    drone.add_supported_prop(HydraClassProp("http://schema.org/model", "model", True, False, False))
+    drone.add_supported_prop(HydraClassProp("http://auto.schema.org/speed", "MaxSpeed", True, False, False))
+    # Operations
+    drone.add_supported_op(HydraClassOp("SubmitStatus",
+                                        "PUT",
+                                        "vocab:Status",
+                                        None,
+                                        [{"code": 200, "description": "Drone Status updated"}]))
+    drone.add_supported_op(HydraClassOp("GetDrone",
+                                        "GET",
+                                        None,
+                                        "vocab:Drone",
+                                        [{"code": 200, "description": "Drone Returned"}]))
+
+    command = HydraClass("Command", "Command", "Class for drone commands")
+    command.add_supported_prop(HydraClassProp("http://schema.org/UpdateAction", "Update", False, True, False))
+    command.add_supported_prop(HydraClassProp("vocab:Status", "Status", False, False, False))
```

Comment:
I don't think commands should send status objects. Status includes things like battery, sensors, sensorStatus and model. I think commands should only include `destination` and `speed`.